nexus-repository-composer/src/main/java/org/sonatype/nexus/repository/composer/internal/ComposerHostedFacetImpl.java

```diff
@@ -84,7 +84,8 @@ public Content getProviderJson(final String vendor, final String project) throws
   @Override
   @TransactionalTouchMetadata
   public Content getPackageJson(final String vendor, final String project) throws IOException {
-    return content().get(ComposerPathUtils.buildPackagePath(vendor, project));
+    Content content = content().get(ComposerPathUtils.buildPackagePath(vendor, project));
+    return content != null ? content : content().get(ComposerPathUtils.buildProviderPath(vendor, project));
```

Comment:
should we add a comment to signal that this is compatibility layer to support composer v1 packages?

Reply:
I will leave a comment there.